x-pack/legacy/plugins/siem/public/components/header_page_new/index.tsx

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiBadge,
+  EuiBetaBadge,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiButtonIcon,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiProgress,
+  EuiTitle,
+} from '@elastic/eui';
+import React from 'react';
+import styled, { css } from 'styled-components';
+
+import { DefaultDraggable } from '../draggables';
+import { LinkIcon, LinkIconProps } from '../link_icon';
+import { Subtitle, SubtitleProps } from '../subtitle';
+import * as i18n from './translations';
+
+interface HeaderProps {
+  border?: boolean;
+  isLoading?: boolean;
+}
+
+const Header = styled.header.attrs({
+  className: 'siemHeaderPage',
+})<HeaderProps>`
+  ${({ border, theme }) => css`
+    margin-bottom: ${theme.eui.euiSizeL};
+
+    ${border &&
+      css`
+        border-bottom: ${theme.eui.euiBorderThin};
+        padding-bottom: ${theme.eui.paddingSizes.l};
+        .euiProgress {
+          top: ${theme.eui.paddingSizes.l};
+        }
+      `}
+  `}
+`;
+Header.displayName = 'Header';
+
+const FlexItem = styled(EuiFlexItem)`
+  display: block;
+`;
+FlexItem.displayName = 'FlexItem';
+
+const LinkBack = styled.div.attrs({
+  className: 'siemHeaderPage__linkBack',
+})`
+  ${({ theme }) => css`
+    font-size: ${theme.eui.euiFontSizeXS};
+    line-height: ${theme.eui.euiLineHeight};
+    margin-bottom: ${theme.eui.euiSizeS};
+  `}
+`;
+LinkBack.displayName = 'LinkBack';
+
+const Badge = styled(EuiBadge)`
+  letter-spacing: 0;
+`;
+Badge.displayName = 'Badge';
+
+const StyledEuiBetaBadge = styled(EuiBetaBadge)`
+  vertical-align: middle;
+`;
+
+StyledEuiBetaBadge.displayName = 'StyledEuiBetaBadge';
+
+const StyledEuiButtonIcon = styled(EuiButtonIcon)`
+  ${({ theme }) => css`
+    margin-left: ${theme.eui.euiSize};
+  `}
+`;
+
+StyledEuiButtonIcon.displayName = 'StyledEuiButtonIcon';
+
+interface BackOptions {
+  href: LinkIconProps['href'];
+  text: LinkIconProps['children'];
+}
+
+interface BadgeOptions {
+  beta?: boolean;
+  text: string;
+  tooltip?: string;
+}
+
+interface DraggableArguments {
+  field: string;
+  value: string;
+}
+interface IconAction {
+  'aria-label': string;
+  iconType: string;
+  onChange: (a: string) => void;
+  onClick: (b: boolean) => void;
```

Comment:
why do we need a boolean?

Reply:
poorly named, i will clean this up in the refactor (combining `header_page_new` back into `header_page`)